src/ex7/client/src/components/todos-list.js

```diff
@@ -1,27 +1,23 @@
 import React from "react";
-import PropTypes from "prop-types";
-import { TodoElement } from "./todo-element.js";
 import { NoTodosPlaceholder } from "./no-todos-placeholder.js";
-import { Footer } from "./footer.js";
+import FilterConnector from "../connectors/filter-connector";
+import TodoElementConnector from "../connectors/todo-element-connector";
+import FooterConnector from "../connectors/footer-connector.js";
 import "../styles/todo-list.css";
 
-export function TodosList(props) {
-  const {todos, updateTodos, loaded} = props;
+export function TodosList({ loaded, todos, filterFunction }) {
+  const amount = todos.length;
```

Comment:
let's move the filtered todos logic to a separate service. Also, remove it from the state, and use use-selector instead for caching